src/java.management.rmi/share/classes/javax/management/remote/rmi/RMIConnection.java

```diff
@@ -956,16 +956,17 @@ public void removeNotificationListener(ObjectName name,
      * @param filters an array of marshalled representations of the
      * <code>NotificationFilters</code>.  Elements of this array can
      * be null.
-     * @param delegationSubjects must be {@code null}.
+     * @param delegationSubjects should be {@code null}, but a non-null
+     * array is accepted for compatibilty reasons, which must not contain
+     * any entries.
      *
      * @return an array of <code>listenerIDs</code> identifying the
      * local listeners.  This array has the same number of elements as
      * the parameters.
      *
      * @throws IllegalArgumentException if <code>names</code> or
      * <code>filters</code> is null, or if <code>names</code> contains
-     * a null element, or if the three arrays do not all have the same
-     * size.
+     * a null element, or if these two arrays do not have the same size.
```

Comment:
Was this actually an oversight in the previous change to remove subject delegation? When `delegationSubjects` is null, then the 3 arrays are never going to be the same size.

Reply:
Yes, this is an oversight from the previous change.